docs/protocol/off-chain/app-sessions.mdx

```diff
@@ -0,0 +1,778 @@
+
+### Response
+
+| Parameter | Type | Description | Format/Structure | Example | Notes |
+|-----------|------|-------------|------------------|---------|-------|
+| `app_session_id` | string | Session identifier (echoed) | - | - | - |
+| `status` | string | Final status | Value: "closed" | `"closed"` | - |
+| `final_allocations` | Allocation[] | Confirmed final distribution | - | - | - |
+| `final_session_data` | string | Confirmed final application state | - | - | - |
+| `distributions` | object | How funds were distributed back to unified balances | Map of participant to amounts returned | `{"0xAlice": {"usdc": "180.0"}, "0xBob": {"usdc": "15.0"}, "0xJudge": {"usdc": "5.0"}}` | - |
+| `closed_at` | number | Unix timestamp of closure | - | `1699125678901` | - |
+| `duration` | number | Session duration in milliseconds | - | `3600000` (1 hour) | - |
+| `total_states` | number | Total number of state updates during session | - | `42` | - |
+
+---
+
+## Fund Distribution on Closure
+
+When an <Tooltip content="Off-chain channels built on top of payment channels, intended to be used by app developers to enable application-specific interactions and transactions without touching the blockchain">app session</Tooltip> closes, funds return to <Tooltip content="An entity (identified by a wallet address) that is part of a channel">participants'</Tooltip> <Tooltip content="An abstraction that aggregates a user's funds across multiple blockchain networks, managed by Clearnode">unified balances</Tooltip>:
+
+```
+Before Closure:
+  Alice's Unified Balance:
+    Total: 200 USDC
+    Available: 100 USDC
+    Locked: 100 USDC (in app session 0x98765...)
+
+App Session 0x98765:
+  Alice: 100 USDC
+  Bob: 100 USDC
+  Total: 200 USDC
+
+Close with final allocations:
+  Alice: 180 USDC
+  Bob: 20 USDC
+
+After Closure:
+  Alice's Unified Balance:
+    Total: 280 USDC (200 + 80 won)
+    Available: 280 USDC
+    Locked: 0 USDC
+    
+  Bob's Unified Balance:
+    Total: 20 USDC (100 - 80 lost)
+    Available: 20 USDC
+    Locked: 0 USDC (was 100)
+```
+
+### Allocation Rules
+
+1. **Must Sum to Total**:
+   - `sum(final_allocations) MUST equal sum(current_allocations)`
+   - Clearnode validates this; cannot create or destroy funds during close
+
+2. **Can Be Zero**:
+   - <Tooltip content="An entity (identified by a wallet address) that is part of a channel">Participants</Tooltip> can receive zero in final allocation (lost everything)
+   - Example: Losing player in a winner-takes-all game
+
+3. **Must Account for All Participants**:
+   - Every <Tooltip content="An entity (identified by a wallet address) that is part of a channel">participant</Tooltip> in definition should have an entry (even if zero)
+   - Omitted <Tooltip content="An entity (identified by a wallet address) that is part of a channel">participants</Tooltip> are treated as receiving zero
```

Comment:
It sounds to me that these points are conflicting, and the first one is not correct, right @dimast-x ?

Reply:
Clarified it further
[ff21dbaf86eb4b18cf046b88a721fd96a2e559f6](https://github.com/layer-3/docs/pull/69/commits/ff21dbaf86eb4b18cf046b88a721fd96a2e559f6)